packages/stg-evm-v2/devtools/src/credit-messaging/state.ts

```diff
@@ -0,0 +1,138 @@
+                    }
+                    return edge
+                }
+            )
+        )
+    }
+
+    const graphBuilder = new OmniGraphBuilder<CreditMessagingNodeConfig, CreditMessagingEdgeConfig>()
+
+    return Promise.all(nodePromises)
+        .then((nodes) => graphBuilder.addNodes(...nodes))
+        .then(() => Promise.all(edgePromises))
+        .then((edges) => graphBuilder.addEdges(...edges))
+        .then(() => graphBuilder.graph)
+}
+
+async function getNodeFromChain(
+    point: OmniPoint,
+    existingConfig: CreditMessagingNodeConfig
+): Promise<OmniNode<CreditMessagingNodeConfig>> {
+    const logger = createModuleLogger(`State Loader for CreditMessaging`)
+    const createSdk = createCreditMessagingFactory(createConnectedContractFactory())
+    const sdk = await createSdk(point)
+
+    logger.debug(`Querying ${getNetworkNameForEid(point.eid)}...`)
+
+    const onChainAssets: MessagingAssetConfig = !existingConfig.assets
+        ? {}
+        : Object.fromEntries(
+              (
+                  await Promise.all(
+                      Object.values(existingConfig.assets).map(async (assetId) => [
+                          await sdk.getAsset(assetId),
+                          assetId,
+                      ])
+                  )
+              ).filter(([assetAddress, _assetId]) => assetAddress)
+          )
+
+    const config: CreditMessagingNodeConfig = {
+        assets: onChainAssets,
+        planner: (await sdk.getPlanner()) ?? '0x',
+        maxAssetId: (await sdk.getMaxAssetId()) ?? 0,
+    }
+
+    return { point, config }
+}
+
+async function getEdgeFromChain(
+    vector: OmniVector,
+    existingConfig: CreditMessagingEdgeConfig
+): Promise<OmniEdge<CreditMessagingEdgeConfig>> {
+    const logger = createModuleLogger(`State Loader for CreditMessaging`)
+    const createSdk = createCreditMessagingFactory(createConnectedContractFactory())
+    const createOappSdk = createOAppFactory(createConnectedContractFactory())
+    const sdk = await createSdk(vector.from)
+    const oappSdk = await createOappSdk(vector.from)
+    const endpointSdk = await oappSdk.getEndpointSDK()
+
+    logger.debug(`Querying edges on ${getNetworkNameForEid(vector.from.eid)}...`)
+
+    const [sendLibrary, recvLibrary] = await Promise.all([
+        endpointSdk.getSendLibrary(vector.from.address, vector.to.eid),
+        endpointSdk.getReceiveLibrary(vector.from.address, vector.to.eid),
+    ])
+
+    const [executorConfig, sendUlnConfig] = sendLibrary
+        ? await Promise.all([
+              endpointSdk.getExecutorConfig(vector.from.address, sendLibrary ?? '0x', vector.to.eid),
+              endpointSdk.getUlnConfig(vector.from.address, sendLibrary ?? '0x', vector.to.eid),
+          ])
+        : [undefined, undefined]
+
+    const recvUlnConfig = recvLibrary
+        ? await endpointSdk.getUlnConfig(vector.from.address, recvLibrary[0] ?? '0x', vector.to.eid)
```

Comment:
`0x` is not a valid address value. Also if you want the `recvLibrary`'s address only, you can do:

```typescript
const [sendLibrary, [recvLibrary]] = await Promise.all([
      endpointSdk.getSendLibrary(vector.from.address, vector.to.eid),
      endpointSdk.getReceiveLibrary(vector.from.address, vector.to.eid),
])
```